source/III.VC.CLEO/Fxt.cpp

```diff
@@ -24,6 +24,40 @@ CustomTextEntry::~CustomTextEntry()
 		delete m_pText;
 }
 
+static uint8_t get_utf8_bytes(uint8_t utf8)
+{
+	for (uint8_t i = 0; i < 6; i++) {
+		if ((utf8 & (0x80 >> i)) == 0) {
+			return i == 0 ? 1 : i;
+		}
+	}
+	return 1;
+}
+
+uint16_t CustomText::convert_utf8_to_utf16(uint8_t* utf8, uint16_t utf8_len, uint16_t* utf16, uint16_t utf16_len)
+{
+	uint16_t length = 0;
+	uint8_t bytes = 0;
+
+	for (uint16_t i = 0; i < utf8_len && length < utf16_len; i++, length++) {
+		bytes = get_utf8_bytes(utf8[i]);
+
+		if (bytes > 1) {
+			utf16[length] = utf8[i] & (0xFF >> (bytes + 1));
```

Comment:
Many magic in single line. How about creating variable named "mask", then using in. Comments would be nice too.